src/lib/RuleResultRecombinator.ts

```diff
@@ -24,135 +22,144 @@ export class RuleResultRecombinator {
-		const columns = rowStrings.shift().split(',').map(v => {
-			return v.split('"')[1];
-		});
-		// Next, we need to turn each row into a JSON.
-		const rowJsons = rowStrings.map(rowStr => {
-			const rowVals = rowStr.split(',');
-			const rowJson = {};
-			columns.forEach((col, idx) => {
-				rowJson[col] = rowVals[idx];
-			});
-			return rowJson;
-		});
-		// Turn our JSON into a string so we can pass it back up through and parse it when we need.
-		return JSON.stringify({columns, rows: rowJsons});
-	}
+	private static constructXml(results: RuleResult[]): string {
+		let resultXml = ``;
 
-	private static pmdToCsv(pmdResults: string): string {
 		// If the results were just an empty string, we can return it.
-		if (pmdResults === '') {
-			return '';
+		if (results.length === 0) {
+			return resultXml;
 		}
 
-		const engine = PmdEngine.NAME;
-
-		// TODO: This logic will definitely need to get more sophisticated so we can create a CSV with the columns that we
-		//  ultimately want. But for now, we'll take the coward's way out and convert the XML into an approximation of the CSV
-		//  that would have been produced by PMD if we'd requested it.
-		// PMD's results are given to us as an XML contained in a string. To turn the results into a CSV, first we'll need
-		// to turn the XML into a JSON that we can easily iterate through.
-		const pmdJson = xml2js(pmdResults, {compact: false, ignoreDeclaration: true});
-		// We'll gradually build our CSV, starting with these columns.
-		let results = '"Problem","File","Priority","Line","Description","Category","Rule","Engine"\n';
 		let problemCount = 0;
 
 		// The top-level Element just has an 'property', which is a singleton list containing the 'pmd' Element, whose own
 		// 'elements' property is a list of 'file' Elements.
-		const fileElements = pmdJson.elements[0].elements;
-		for (const fileElement of fileElements) {
-			const fileName = fileElement.attributes.name;
+		for (const result of results) {
+			const from = process.cwd();
+			const fileName = path.relative(from, result.fileName);
 			// Each 'file' Element should contain in its 'elements' property a list of 'violation' Elements describing the ways
 			// rules were violated.
-			const violations = fileElement.elements;
-			for (const violation of violations) {
-				const attrs = violation.attributes;
-				// The error message for the violation is a 'text' Element below it. And since we are creating
-				// CSVs, make sure we escape any commas in our violation messages.  Just replace with semi-colon.
-				const msg = violation.elements[0].text.trim().replace(",", ";");
-				const row = [++problemCount, fileName, attrs.priority, attrs.beginline, msg, attrs.ruleset, attrs.rule, engine];
-				results += '"' + row.join('","') + '"\n';
+			let violations = '';
+			for (const v of result.violations) {
+				problemCount++;
+				violations += `
```

Comment:
There should be a comment here explaining that we're essentially turning the violation objects into an XML of some kind.